library/src/main/java/com/chuckerteam/chucker/internal/support/TeeSource.kt

```diff
@@ -90,10 +90,11 @@ internal class TeeSource(
     interface Callback {
         /**
          * Called when the upstream was closed. All read bytes are copied to the [file].
-         * This does not mean that the content of a [file] is valid. Only that the user
-         * is done with the reading process.
+         * This does not mean that the content of the [file] is valid. Only that the user
+         * is done with the reading process. [totalBytesRead] is the exact amount of data
+         * that the client downloaded even if the [file] is corrupted.
          */
-        fun onClosed(file: File)
+        fun onClosed(file: File, totalBytesRead: Long)
```

Comment:
Let's invert the parameter order either here or in the onFailure:

```kotlin
fun onClosed(file: File, totalBytesRead: Long)
fun onFailure(file: File, exception: IOException)
```

File should always be either the first or the last param.